pkgs/by-name/pa/panda3ds/package.nix

```diff
@@ -0,0 +1,199 @@
+    [
+      SDL2
+      fdk_aac
+      boost
+    ]
+    ++ lib.optionals withQtGui (
+      [ kdePackages.qtbase ]
+      ++ (lib.optional (
+        lib.versionAtLeast kdePackages.qtbase.version "6" && stdenv.hostPlatform.isLinux && withQtWayland
+      ) kdePackages.qtwayland)
+    )
+    ++ lib.optionals withLuaJIT [
+      luajit
+      libuv
+    ]
+    ++ lib.optionals withDiscordRPC [
+      discord-rpc
+      rapidjson
+    ]
+    ++ lib.optionals withVulkan [
+      vulkan-headers
+      vulkan-loader
+      glslang
+    ]
+    ++ lib.optionals withOpenGL [ libGL ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      xorg.libX11
+      wayland
+    ];
+
+  # delete unused third_party deps
+  postUnpack = ''
+    rm -rf third_party/{SDL2,discord-rpc,fdk-aac,LuaJIT,boost}
+  '';
+
+  # fix missing ir/var/empty/constant_propagation_pass.cpp
+  postPatch = ''
+    mkdir -p third_party/dynarmic/src/dynarmic/ir/var
+    ln -s ../opt third_party/dynarmic/src/dynarmic/ir/var/empty
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # main binary
+    install -Dm755 /build/source/build/Alber $out/bin/Alber
+
+    # desktop file
+    install -Dm444 $src/.github/Alber.desktop $out/share/applications/Alber.desktop
+
+    # icons
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+    for size in 16 32 48 64 72 96 128 192 256 512 1024; do
+      mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+      magick convert \
+        -resize "$size"x"$size" \
+        -alpha on \
+        -background none \
+        $src/docs/img/Alber.png \
+        $out/share/icons/hicolor/"$size"x"$size"/apps/Alber.png
+    done
+
+    runHook postInstall
+  '';
+
+  postInstall =
+    let
+      libs = lib.makeLibraryPath (
+        lib.optionals withOpenGL [ libGL ]
+        ++ lib.optionals withVulkan [ vulkan-loader ]
+        ++ lib.optionals withDiscordRPC [ discord-rpc ]
+        ++ lib.optionals stdenv.hostPlatform.isLinux [ pulseaudio ]
+      );
+    in
+    if withQtGui then
+      ''
+        qtWrapperArgs+=(
+          --prefix LD_LIBRARY_PATH : ${libs}
+      ''
+      + (lib.optionalString (stdenv.hostPlatform.isLinux && !withQtWayland) ''
+        --set QT_QPA_PLATFORM "xcb"
+      '')
+      + ")"
+    else
+      ''
+        wrapProgram "$out/bin/Alber" \
+          --prefix LD_LIBRARY_PATH : ${libs}
+      '';
+
+  cmakeFlags = [
+    (lib.cmakeBool "ENABLE_USER_BUILD" true)
+    (lib.cmakeBool "DISABLE_PANIC_DEV" true)
+    (lib.cmakeBool "ENABLE_RENDERDOC_API" withRenderdoc)
+    (lib.cmakeBool "ENABLE_METAL" withMetal)
+    (lib.cmakeBool "ENABLE_VULKAN" withVulkan)
+    (lib.cmakeBool "ENABLE_OPENGL" withOpenGL)
+    (lib.cmakeBool "ENABLE_LUAJIT" withLuaJIT)
+    (lib.cmakeBool "ENABLE_DISCORD_RPC" withDiscordRPC)
+    (lib.cmakeBool "ENABLE_QT_GUI" withQtGui)
+    (lib.cmakeBool "DISABLE_SSE4" (!sse4Support))
+    (lib.cmakeBool "ENABLE_LTO" true)
+    (lib.cmakeBool "USE_SYSTEM_SDL2" true)
+    # "-DENABLE_TESTS=ON"
```

Comment:
Why can't we run the tests? Do they need ROMs?

Reply:
That would only *build* them (it's using the Catch2 test framework)
Just haven't got around to figuring out how to actually run them yet... 
(or if they would actually work... )

Reply:
Gotcha, looking into that would be very cool if they are able to be run

Reply:
They are broken. Won't even compile

```
       > /build/source/tests/shader.cpp: In function 'std::unique_ptr<PICAShader> assembleVertexShader(std::initializer_list<nihstro::InlineAsm>)':
       > /build/source/tests/shader.cpp:35:20: error: 'class PICAShader' has no member named 'finalize'
       >    35 |         newShader->finalize();
       >       |                    ^~~~~~~~
```

Reply:
Fixed on upstream